torch/_inductor/codecache.py

```diff
@@ -402,8 +409,24 @@ def get_include_and_linking_paths(
         # This approach allows us to only pay for what we use.
         ipaths = cpp_extension.include_paths() + [sysconfig.get_path("include")]
         lpaths = []
-        libs = ["gomp"]
         macros = ""
+        if sys.platform == "darwin":
+            # GNU OpenMP generally is not available on MacOS
+            # There is either Intel OpenMP(for x86) or LLVM OpenMP (for both x86 and arm64)
+            libs = ["omp"]
+            if os.getenv("CONDA_PREFIX") is not None:
```

Comment:
I think we should give users more feedback related to this as I don't thing most users use conda.
In particular should we generally recommend people use conda if they want to use torch.compile? Should we do that on macos at least?
If they don't do that, then they are reponsible for setting up the right global deps.

If that is our stance, I think we should mention that in the doc + have a warning.

Reply:
Our `torch.compile` CPU story is somewhat confusing:
- It plainly does not work on Windows
- Until this PR I don't know how one can make it work on MacOS (I could not find how to install gnu OpenMP on MacOS, not even sure if this is possible)
- It works on Linux only if openmp is installed 

But yeah, we probably should bundle `omp.h` into `torch/include` and use it from there by default, will probably do that in followup PR.